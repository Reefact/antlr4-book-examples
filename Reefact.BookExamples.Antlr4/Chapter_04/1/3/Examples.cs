#region Usings declarations

using System.Text;

using Antlr4.Runtime;

using ApprovalTests;
using ApprovalTests.Reporters;

using Reefact.BookExamples.Antlr4.Chapter_04._1._2;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_04._1._3 {

    [UseReporter(typeof(CustomReporter))]
    public class Examples {

        [Fact]
        public void lisp_style_tree_example() {
            // Setup
            StringBuilder example = new();
            example.AppendLine("(1+2");
            example.AppendLine("3");
            AntlrInputStream inputStream = AntlrInputStreamReader.Read(example);
            // Exercise
            GRun   grun          = GRun.Read(inputStream);
            string lispStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(lispStyleTree);
        }

        [Fact]
        [GraphicalTree("ParseTree1.svg")]
        public void mermaid_style_tree_example() {
            // Setup
            StringBuilder example = new();
            example.AppendLine("(1+2");
            example.AppendLine("34*69");
            AntlrInputStream inputStream = AntlrInputStreamReader.Read(example);
            // Exercise
            GRun   grun             = GRun.Read(inputStream);
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

    }

}