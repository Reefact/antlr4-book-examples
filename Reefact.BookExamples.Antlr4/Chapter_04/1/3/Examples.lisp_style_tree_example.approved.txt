line 1:4 missing ')' at '\r\n'
(prog (stat (expr ( (expr (expr 1) + (expr 2)) <missing ')'>) \r\n) (stat (expr 3) \r\n))